src/components/structures/MatrixChat.js

```diff
@@ -584,6 +588,44 @@ module.exports = React.createClass({
     _onLoadCompleted: function() {
         this.props.onLoadCompleted();
         this.setState({loading: false});
+
+        // set up the right theme.
+        // XXX: this will temporarily flicker the wrong CSS.
+        dis.dispatch({
+            action: 'set_theme',
+            value: UserSettingsStore.getSyncedSetting('theme')
+        });
+    },
+
+    /**
+     * Called whenever someone changes the theme
+     */
+    _onSetTheme: function(theme) {
```

Comment:
yes, but I'd like to suggest it gets done as part of the great re-layering of react-sdk & riot-web. the theming is already split over both libs, whereas in practice it should probably somehow be limited to the application layer. keeping it in the dumping ground for now avoids having to spend time threading things around the place only to decide to change it again in the nearish future when we get that far.